gapis/api/vulkan/frame_loop.go

```diff
@@ -1360,6 +1414,273 @@ func (f *frameLoop) resetResources(ctx context.Context, stateBuilder *stateBuild
 	return nil
 }
 
+func (f *frameLoop) resetInstances(ctx context.Context, stateBuilder *stateBuilder) error {
+
+	// For every Instance that we need to destroy at the end of the loop...
+	for toDestroy := range f.instanceToDestroy {
+		// Write the command to delete the created object
+		instance := GetState(f.loopEndState).instances.Get(toDestroy)
+		stateBuilder.write(stateBuilder.cb.VkDestroyInstance(instance.VulkanHandle(), memory.Nullptr))
+	}
+
+	// For every Instance that we need to create at the end of the loop...
+	for toCreate := range f.instanceToCreate {
+		// Write the commands needed to recreate the destroyed object
+		instance := GetState(f.loopStartState).instances.Get(toCreate)
+		stateBuilder.createInstance(instance.VulkanHandle(), instance)
+	}
+
+	for _, deviceObject := range GetState(f.loopStartState).Devices().All() {
+
+		physicalDevice := deviceObject.PhysicalDevice()
+		physicalDeviceObject := GetState(f.loopStartState).PhysicalDevices().All()[physicalDevice]
+		instance := physicalDeviceObject.Instance()
+
+		if _, ok := f.instanceToCreate[instance]; ok {
+			f.deviceToDestroy[deviceObject.VulkanHandle()] = true
+			f.deviceToCreate[deviceObject.VulkanHandle()] = true
+		}
```

Comment:
Do we need to handle surface here?

Reply:
Good catch. That was in code review while I was doing this work, and I based the list of things to rebuild off the code as checked in to master.